Description/IconEnhancer.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2015 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\TreeBrowserBundle\Description;
+
+use Symfony\Cmf\Component\Resource\Repository\Resource\CmfResource;
+use Symfony\Cmf\Component\Resource\Description\Description;
+use Symfony\Cmf\Component\Resource\Description\DescriptionEnhancerInterface;
+use Puli\Repository\Api\Resource\PuliResource;
+
+class IconEnhancer implements DescriptionEnhancerInterface
+{
+    private $iconMap;
+    private $classMap = array();
+
+    public function __construct(array $iconMap)
+    {
+        $this->iconMap = $iconMap;
+    }
+
+    public function enhance(Description $description)
+    {
+        $class = get_class($description->getResource()->getPayload());
+        if (isset($this->classMap[$class])) {
+            $class = $this->classMap[$class];
+        }
+
+        $description->set('icon', $this->iconMap[$class]);
```

Comment:
for context, I am currently doing this:

![sycms](https://cloud.githubusercontent.com/assets/530801/18148699/3331cdc0-6fd3-11e6-8f57-88d98a2f6ca0.png)

Whereby the "image" is added to the description from any image mappings for the (resource) in the content-type component and subsequently used as the thumbnail via. an imagine resize.